controllers/nodemaintenance_controller.go

```diff
@@ -150,6 +150,7 @@ func (r *NodeMaintenanceReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 
 	updateOwnedLeaseFailed, err := r.obtainLease(node)
 	if err != nil && updateOwnedLeaseFailed {
+		setLastUpdate(instance)
```

Comment:
you are setting last update a bit too often IMHO.
For example here: the only way out of this block is `onReconcileError()`, and there you also set last update. So it can be removed here. The other place is better, because that is just before the status is actually updated.